File: Chapter04.GeneralizationSpecializationAndPolymorphism/C++/Rectangle.cpp
#include "Rectangle.h"

Rectangle::Rectangle() {
    this->width = 0;
    this->height = 0;
}

Rectangle::Rectangle(float w, float h) {
    this->width = w;
    this->height = h;
}

void Rectangle::input(std::istream& inDev) {
    inDev >> this->width >> this->height;
}

float Rectangle::area() {
    return this->width * this->height;
}

File: Chapter04.GeneralizationSpecializationAndPolymorphism/Java/Rectangle.java
//use keyword "extends" to define a child class
class Rectangle extends Figure {
    private float width, height;

    public Rectangle(float w, float h) {
        width = w;
        height = h;
    }

    //String is not considered as a keyword of language,
    //it doesn't have color in text editor
    public String className() {
        return "Rectangle";
    }

    public float area() {
        return width * height; 
    }
}